lib/core/terraform_config/generator.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module TerraformConfig
+  class Generator
+    attr_reader :config, :template
+
+    def initialize(config:, template:)
+      @config = config
+      @template = template
+    end
+
+    def filename
+      case template["kind"]
+      when "gvc"
+        "gvc.tf"
+      when "identity"
+        "identities.tf"
+      else
+        raise "Unsupported template kind - #{template['kind']}"
+      end
+    end
+
+    def tf_config
+      case template["kind"]
+      when "gvc"
+        gvc_config
+      when "identity"
+        identity_config
+      else
+        raise "Unsupported template kind - #{template['kind']}"
+      end
+    end
+
+    private
+
+    # rubocop:disable Metrics/MethodLength
```

Comment:
Nit: This can be moved to the end of line 37. That way we don't have to enable the rule again later, it gets disabled only for the block.